### PR TITLE
Course enrollment: Student Joining Classroom by Providen Access Code, Dynamic Card view . Student Classroom Route Links

### PR DESCRIPTION
**Pull request by zuizihad**
**Whats Done**

- Students can join classrooms by providen classroom access code of implemented classroom by teacher.
- Implemented Dynamic Classroom Cards after each post request for accessing  of a new classroom without reloading the current page
- Added routes with blank target for each classroom with sid and cid
- Implemented functionality for random colors of each card

localhost_5000_Students___init___sid=c3e8d7569f06430fa3427896489a96d2

localhost_5000_Students___classrooms____sid=c3e8d7569f06430fa3427896489a96d2

localhost_5000_Students___classrooms____sid=341ba9c830594f4899a98007dac7f5b4

### CODE REVIEW

**Comment by shunjid** [comment]
@jinnatul : Tested with clean database file. Primarily looks like all the functionalities are working well

File: src/Views/Shared/_LayoutStudent.cshtml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    <!-- Fonts & Icons Required To Enhance The Look -->
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Roboto+Mono&display=swap" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Product+Sans:400&amp;lang=en" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Google+Sans:400,500&amp;lang=en" rel="stylesheet">
+    
+    <!-- Stylesheets that are going to work in Development Environment -->
+    <environment include="Development">
+        <link rel="stylesheet" href="~/app/lib/stylesheets/materialize.css" >
+        <link rel="stylesheet" href="~/app/src/stylesheets/__initTeacherLayout__.css" >
```
Comment (shunjid) [inline src/Views/Shared/_LayoutStudent.cshtml]: Why not a new file_initStudentLayout.css ?????????

File: src/wwwroot/app/src/scripts/__ClassroomJoin.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+function getClassRoom() {
+
+    var classrooms = $('#classRooms');
+    console.log("sid is : " + tempsid);
+    console.log("classroom is : " + classrooms);
+
+    $.get('/Students/__getClassRoom___', {sid : tempsid}, function (res) {
+        if(res.length == 0){
+            var modalForCreateNewClassroom = "<h5 class=\"center-align\">You have no classroom yet</h5>\n" +
```
Comment (shunjid) [inline src/wwwroot/app/src/scripts/__ClassroomJoin.js]: Change the variable name. It should be modalToJoinNewClassroom. Carefully change this variabe in other places where used. Try finding the word with Ctrl + F

File: src/Controllers/StudentsController.cs [hunk 3]
```diff
@@ -58,7 +62,145 @@ public async Task<IActionResult> __init__(string sid)
...
+        
+        public async Task<JsonResult>__getClassRoom___(string sid)
+        {
+            var classroomsOfStudent =
+                await _context.StudentEnrollments
+                    .Include(c => c.Student)
+                    .Include(c => c.Student.Account)
+                    .Include(c => c.Classroom)
+                    .Include(c => c.Classroom.Teacher.Account)
+                    .Where(tec => tec.Student.Account.UserId == sid)
```
Comment (jinnatul) [inline src/Controllers/StudentsController.cs]: why you use tec in where condition?
we know this name is not an important issue but we already use tec in the teacher(create classroom). so it's a confusing name, you can try this std/stu. then we can easily understand this :)

File: src/Controllers/StudentsController.cs [hunk 4]
```diff
@@ -58,7 +62,145 @@ public async Task<IActionResult> __init__(string sid)
...
+                    .ToListAsync();
+            
+                        
+            return Json(classroomsOfStudent);
+        }
+        
+        private bool StudentExists(int id)
+        {
+            return _context.Students.Any(e => e.Serial == id);
```
Comment (jinnatul) [inline src/Controllers/StudentsController.cs]: try to you meaningful word/ variable name (Ex: e, i, a, ....)
next time you can easily understand you code and this meaningful word provides you, coding documentation.

File: src/Views/Students/__init__.cshtml [hunk 5]
```diff
@@ -1,3 +1,98 @@
...
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+
+<div id="_formModal" class="modal">
+    <div class="modal-content">
+      <form id="_formTeacherInfo" method="post">
```
Comment (jinnatul) [inline src/Views/Students/__init__.cshtml]: why you use form id="_formTeacherInfo"? 
It's Teachers/Students view?

File: src/wwwroot/app/src/scripts/__ClassroomJoin.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
+var tempsid = $('#tempsid').val();
+
+var colorArray = [
+    "red darken-1", "teal darken-1", "indigo darken-1", "blue-grey darken-1",
+    "purple darken-3", "blue lighten-1", "blue accent-2", "light-blue"
+];
+
+$(document).ready(function () {
+    getClassRoom();
```
Comment (jinnatul) [inline src/wwwroot/app/src/scripts/__ClassroomJoin.js]: we already use this method for teachers. getClassRoom();
you can try to use getClassRoomForStudent();

**Comment by shunjid** [comment]
@jinnatul : Please, List down what we need to change in his codes. In the comment & fill-up your review report.

**Pull request merged**